dlt/extract/validation.py

```diff
@@ -0,0 +1,35 @@
+from typing import Optional, Protocol, TypeVar, Generic, Type, Union, Any
+
+try:
+    from pydantic import BaseModel as PydanticBaseModel, ValidationError as PydanticValidationError
+except ModuleNotFoundError:
+    PydanticBaseModel = None  # type: ignore[misc]
+
+from dlt.extract.exceptions import ValidationError
+from dlt.common.typing import TDataItem
+from dlt.common.schema.typing import TAnySchemaColumns, ColumnValidator
+from dlt.extract.typing import TTableHintTemplate
+
+
+_TPydanticModel = TypeVar("_TPydanticModel", bound=PydanticBaseModel)
+
+
+class PydanticValidator(ColumnValidator, Generic[_TPydanticModel]):
+    model: Type[_TPydanticModel]
+    def __init__(self, model: Type[_TPydanticModel]) -> None:
```

Comment:
how does it work without implementing `bind`? something must be wrong because bind must be called on each transform step just before extract on the pipe starts. could you please check it out? if the reason is that I check some kind of base class somewhere then let's create a base class with `bind` for all transforms. this will be very useful

Reply:
Hmm yeah you mean this? https://github.com/dlt-hub/dlt/blob/sthor%2Fpydantic-validation/dlt/extract/pipe.py#L261-L264  
I guess it would be best to make this a subclass of `ItemTransform`?